src/InteractiveWindow/Editor/InteractiveWindow.UIThreadOnly.cs

```diff
@@ -474,12 +474,36 @@ public void InsertCode(string text)
                 }
                 else
                 {
-                    if (!TextView.Selection.IsEmpty)
+                    using (var transaction = UndoHistory_TestOnly.CreateTransaction(InteractiveWindowResources.TypeChar))
```

Comment:
I agree. We generally do not change the name if we are exposing things for testing purposes. It is confusing. We should just put a comment where we are exposing this to indicate that it is exposed for testing purposes only.